prophy/tests/test_get_descriptor.py

```diff
@@ -0,0 +1,107 @@
+import prophy
+import pytest
+
+@pytest.fixture(scope = 'session')
+def Struct():
+    class Struct(prophy.with_metaclass(prophy.struct_generator, prophy.struct)):
+        _descriptor = [("x", prophy.u32),
+                       ("y", prophy.u32)]
+    return Struct
+
+@pytest.fixture(scope = 'session')
+def Union(Struct):
+    class Union(prophy.with_metaclass(prophy.union_generator, prophy.union)):
+        _descriptor = [("a", prophy.u16, 0),
+                       ("b", prophy.u32, 1),
+                       ("c", Struct, 2)]
+    return Union
+
+
+@pytest.fixture(scope = 'session')
+def NestedStruct(Struct):
+    class NestedStruct(prophy.with_metaclass(prophy.struct_generator, prophy.struct)):
+        _descriptor = [("a", Struct),
+                       ("b", Struct)]
+    return NestedStruct
+
+@pytest.fixture(scope = 'session')
+def DeeplyNestedStruct(NestedStruct, Struct):
+    class DeeplyNestedStruct(prophy.with_metaclass(prophy.struct_generator, prophy.struct)):
+        _descriptor = [("m", NestedStruct),
+                       ("n", Struct),
+                       ("o", prophy.u32)]
+    return DeeplyNestedStruct
+
+@pytest.fixture(scope = 'session')
+def ComplicatedStruct(NestedStruct, Struct, Union):
+    class ComplicatedStruct(prophy.with_metaclass(prophy.struct_generator, prophy.struct)):
+        _descriptor = [("a", NestedStruct),
+                       ("b", prophy.u32),
+                       ("c", prophy.array(Struct, size = 2)),
+                       ("d", prophy.array(Union, size = 2))]
+    return ComplicatedStruct
+
+
+
+def test_kind_constant():
+
+    with pytest.raises(ValueError) as e:
+        prophy.kind.kind.STRUCT = 10
+    assert "Cannot assign to a constant." == str(e.value)
+
+
+def test_struct_instance_get_descriptor(DeeplyNestedStruct,NestedStruct,Struct):
+
+  fields = DeeplyNestedStruct().get_descriptor()
```

Comment:
Please change code formatting to be consistent with this file and remaining files in this test suite:
4 spaces as an indent.